doc/compiler/il/SymbolsSymrefsAliasing.md

```diff
@@ -0,0 +1,172 @@
+for optimization. It stores basic information (e.g. data type) needed to determine
+what memory access operations can/must be done.
+
+Different kinds of symbols exist to represent the different kinds of memory access
+requirements. Each kind of symbol stores information that is specific to its kind (type).
+The different symbol kinds are:
+
+* statics
+   * represent static memory address
+   * store the address of the memory location
+* autos
+   * represent variables mapped on the stack
+      * may come from local method variables or JIT allocated stack objects
+   * store stack offset of the variable
+* methods
+   * represent a method (static, virtual)
+* resolved method
+* parameter
+* register mapped
+* label
+* shadow
+   * represent indirection off a base address
+      * e.g. accessing a class field using the classes base address plus an offset
+   * store offset from some base address
+
+#### int shadow
+
+An int shadow represents the common case of an integer type. In the case of Java,
+for example, an int shadow is created for a field or array access.
+
+#### generic int shadow
+
+A generic int shadow is created by the optimizer in (relatively rare) cases when we
+want to refer to a field that wasn't referred to in the bytecodes. If the optimizer
+wants to synthesize a load or store, but doesn't have a symref for it in the original
+program, it can create one and call it a generic int shadow.
+
+For example, escape analysis may want to generate a store to a location within an
+object that hasn't been read in this method in order to initialize the location to zero.
+
+Generic int shadow has different aliasing depending on the context where it is created
+(i.e. it could alias different kinds of other shadows).
+
+### Symrefs
+
+A symbol reference is (or is supposed to be, rather) a wrapper around the different kinds
+of symbols. It encapsulate the additional information required by an optimizing compiler.
+Included information is:
+
+* aliasing
+* owning method (method who's IL created the symbol, which may be different from the
+  current method because of inlining)
+
+All symrefs for a given compilation are stored in the so called "symbol reference table" (symreftab).
+
+Sets of symrefs are almost always represented as bit vectors where each bit specifies
+whether a corresponding symref is a member of the set or not.
+
+## Aliasing
+
+### Basic concept
+
+As was described above, aliasing information is associated with symrefs. However,
+the definition of aliasing used in Testarossa differs from the classical definition
+of aliasing. Specifically, it is used as an indicator of side-effects, rather than to
+mean that multiple names are used to access the same memory. For example, a method 
+symref may alias a static symref, signifying that a call to the method kills
+the value of the static.
+
+### Asymmetric aliasing
+
+This alternative definition of aliasing requires that aliasing relationships
```

Comment:
'requires that aliasing relationships be allowed to be asymmetric' -> 'permits aliasing relationships to be asymmetric'